projects/coordinator/src/main/java/org/batfish/coordinator/WorkMgrService.java

```diff
@@ -156,6 +160,28 @@ public JSONArray configureAnalysis(
       checkClientVersion(clientVersion);
       checkContainerAccessibility(apiKey, containerName);
 
+      Map<String, String> questionsToAdd = new HashMap<>();
+      if (addQuestionsStream != null) {
+        JSONObject jObject = CommonUtil.writeStreamToJSONObject(addQuestionsStream);
+        Iterator<?> keys = jObject.keys();
+        while (keys.hasNext()) {
+          String qName = (String) keys.next();
+          JSONObject qJson;
+          try {
+            qJson = jObject.getJSONObject(qName);
+          } catch (JSONException e) {
+            throw new BatfishException(
+                "Provided questions lack a question named '" + qName + "'", e);
+          }
+          String questionText;
+          try {
+            questionText = qJson.toString(1);
+          } catch (JSONException e) {
+            throw new BatfishException("Failed to convert question JSON to string", e);
+          }
+          questionsToAdd.put(qName, questionText);
+        }
+      }
```

Comment:
1. It looks like this block (Stream -> Map<String, String>) could be moved into a static utility method. This would be much clearer to read, documented, and could be tested in isolation without worrying about file system, etc.

2. Why can we not simply do something like use Jackson to read a Map<String, Object> from the stream?

Reply:
I tried the second approach to use Jackson to convert the input raw question text into a Question object, it gives exception like this : "com.fasterxml.jackson.databind.exc.InvalidTypeIdException: Missing type id when trying to resolve subtype of [simple type, class org.batfish.datamodel.questions.Question]: missing type id property 'class'". 
At Batfish size, we actually [preprocessQuestion](https://github.com/batfish/batfish/blob/master/projects/batfish/src/main/java/org/batfish/main/Batfish.java#L3213) before converting it into a Question object. So I just string to store the question context here.